SVPAlib/src/automata/sra/SRACheckMove.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * SVPAlib
+ * automata
+ * Jul 25, 2018
+ * @author Tiago Ferreira
+ */
+package automata.sra;
+
+import org.sat4j.specs.TimeoutException;
+
+import theory.BooleanAlgebra;
+
+/**
+ * SRACheckMove
+ * @param <P> set of predicates over the domain S
+ * @param <S> domain of the automaton alphabet
+ */
+public class SRACheckMove<P, S> extends SRAMove<P, S> {
+
+	/**
+	 * Constructs an SRA Check Transition that starts from state <code>from</code> and ends at state
+	 * <code>to</code> with input <code>input</code>
+     * Check transitions happen iff the predicate is true and the register mentioned matches the input symbol.
+	 */
+	public SRACheckMove(Integer from, Integer to, P guard, Integer register) {
+		super(from, to, guard, register);
+    }	
+
+	@Override
+	public boolean isSatisfiable(BooleanAlgebra<P, S> boolal) throws TimeoutException {
+        // FIXME: Requires knowledge of the registers at time of transition.
+        //        Possible solution: let guard = guard AND R[register]
+		return boolal.IsSatisfiable(guard);
+	}
+	
+	@Override
+	public boolean isDisjointFrom(SRAMove<P, S> t, BooleanAlgebra<P, S> ba) throws TimeoutException{
+		if (from.equals(t.from)){			
+			SRACheckMove<P, S> ct = (SRACheckMove<P, S>) t;
+            // FIXME: Is having the same register required for disjunction?
```

Comment:
What exactly do we want disjunction to mean with `SRAMove`s? Can they have the same register operation?

Reply:
Indeed, this is tricky. We'll have to think about this a little bit since registers can hold different values. For now just assume that 2 moves are disjoint if: 1) the predicates have empty intersection OR 2) the moves access different registers

Reply:
Okay this sounds like a good start.